SOURCES/plug-late-sr

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+import argparse
+import configparser
+import distutils.util
+import sys
+import time
+import XenAPI
+
+
+PBD_PLUG_RETRY_DELAY = 30  # In seconds.
+
+
+# ------------------------------------------------------------------------------
+
+def strtobool(value):
+    try:
+        return bool(distutils.util.strtobool(value))
+    except ValueError:
+        return False
+
+
+# ------------------------------------------------------------------------------
+
+def get_xapi_session():
+    session = XenAPI.xapi_local()
+    try:
+        session.xenapi.login_with_password('root', '', '', 'plug-late-sr')
+    except Exception as e:
+        raise Exception('Failed to get local Xen API session: {}'.format(e))
+    return session
+
+
+def get_local_host_uuid():
+    with open('/etc/xensource-inventory', 'r') as f:
+        for line in f.readlines():
+            if line.startswith('INSTALLATION_UUID'):
+                return line.split("'")[1]
+
+
+def get_local_host_ref(session):
+    return session.xenapi.host.get_by_uuid(get_local_host_uuid())
+
+
+def get_master_ref(session):
+    pools = session.xenapi.pool.get_all()
+    return session.xenapi.pool.get_master(pools[0])
+
+
+# ------------------------------------------------------------------------------
+
+def plug_pbds(session, sr_uuid, master_ref, sr_ref, retry_delay):
+    pbd_refs = session.xenapi.SR.get_PBDs(sr_ref)
+
+    # Find master PBD and move it before the slaves.
+    master_pbd_index = None
+    for index, pbd_ref in enumerate(pbd_refs):
+        pbd = session.xenapi.PBD.get_record(pbd_ref)
+        if pbd['host'] == master_ref:
+            master_pbd_index = index
+            break
+
+    should_start_vms = True
+    if master_pbd_index is not None:
+        if master_pbd_index != 0:
+            pbd_refs.insert(0, pbd_refs.pop(master_pbd_index))
+        if session.xenapi.PBD.get_currently_attached(pbd_refs[0]):
+            should_start_vms = False
```

Comment:
I prefer to have the master PBD at the beginning of the list. In fact, it should not be mandatory to plug the PBDs with the master first, but it can impact exotic drivers. It was the case for LINSTOR (not anymore), so I think it's a good idea to prevent bad behaviors.

In practice we can plug the PBDs in any order regarding our drivers.